src/function/scalar/string/regexp.cpp

```diff
@@ -203,6 +203,96 @@ static unique_ptr<FunctionData> RegexReplaceBind(ClientContext &context, ScalarF
 	return move(data);
 }
 
+static void RegexExtractFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto &func_expr = (BoundFunctionExpression &)state.expr;
+	auto &info = (RegexpExtractBindData &)*func_expr.bind_info;
+
+	auto &strings = args.data[0];
+	auto &patterns = args.data[1];
+	auto &groups = args.data[2];
+	if (info.constant_pattern) {
+		const RE2 re(duckdb_re2::StringPiece(info.pattern_string));
```

Comment:
Compiling a regex once per vector can still be very expensive, perhaps have a look at `RegexInitLocalState` which is used by the regexp_full_match function to avoid this per-vector compilation in the case of a constant pattern?

Reply:
Good call. I've made it const now.